gammapy/background/models.py

```diff
@@ -71,3 +81,775 @@ def evaluate(self, x, y):
         y = np.asanyarray(y, dtype=float)
         parvals = self.parvals(x)
         return self._evaluate_y(y, parvals)
+
+
+def _make_bin_edges_array(lo, hi):
+    """Make bin edges array from a low values and a high values array.
+
+    TODO: move this function to somewhere else? (i.e. utils?)
+
+    Parameters
+    ----------
+    lo : `~numpy.array`
+        lower boundaries
+    hi : `~numpy.array`
+        higher boundaries
+
+    Returns
+    -------
+    bin_edges : `~numpy.array`
+        array of bin edges as [[low], [high]]
+    """
+    return np.append(lo.flatten(), hi.flatten()[-1:])
+
+
+def make_linear_bin_edges_arrays_from_wcs(w, nbins_x, nbins_y):
+    """Make a 2D linear binning from a WCS object.
+
+    This method gives the correct answer only for linear X, Y binning.
+    The method expects angular quantities in the WCS object.
+    X is identified with WCS axis 1, Y is identified with WCS axis 2.
+    The method needs the number of bins as input, since it is not in
+    the WCS object.
+
+    TODO: move this function to somewhere else? (i.e. utils?)
+
+    Parameters
+    ----------
+    w : `~astropy.wcs.WCS`
+        WCS object describing the bin coordinates
+    nbins_x : `~int`
+        number of bins in X coordinate
+    nbins_y : `~int`
+        number of bins in Y coordinate
+
+    Returns
+    -------
+    bins_x : `~numpy.array` of `~astropy.coordinates.Angle`
+        array with the bin edges for the X coordinate
+    bins_y : `~numpy.array` of `~astropy.coordinates.Angle`
+        array with the bin edges for the Y coordinate
+    """
+    # check number of dimensions
+    if w.wcs.naxis != 2:
+        raise ValueError("Expected exactly 2 dimensions, got {}"
+                         .format(w.wcs.naxis))
+
+    unit_x, unit_y = w.wcs.cunit
+    delta_x, delta_y = w.wcs.cdelt
+    delta_x = Angle(delta_x, unit_x)
+    delta_y = Angle(delta_y, unit_y)
+    bins_x = np.arange(nbins_x + 1)*delta_x
+    bins_y = np.arange(nbins_y + 1)*delta_y
+    # translate bins to correct values according to WCS reference
+    # coordinate start empiricaly determined at pix = 0.5: why 0.5?
```

Comment:
The FITS coordinate convention is that the center of the first pixel is at `(1, 1)`.
That means the start of the image is at `(0.5, 0.5)`.
http://photutils.readthedocs.org/en/latest/photutils/overview.html#coordinate-conventions

So please either remove this question (which looks like a TODO) completely, or add one sentence
```
# In FITS, the edge of the image is at pixel coordinate +0.5.
```